Tools/AP_Periph/can.cpp

```diff
@@ -978,7 +978,7 @@ void AP_Periph_FW::can_mag_update(void)
 #ifdef HAL_PERIPH_ENABLE_MAG
     compass.read();
 #if 1
-    if (compass.get_count() == 0) {
+    if (compass.num_detected() > 0) {
```

Comment:
this has reversed the test - it is meant to run the init() if no sensors have been found